flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableInputFormat.java

```diff
@@ -88,17 +86,23 @@
 	private transient InputFormat mapredInputFormat;
 	private transient HiveTablePartition hiveTablePartition;
 
+	// indices of fields to be returned, with projection applied (if any)
+	// TODO: push projection into underlying input format that supports it
+	private int[] fields;
+
 	public HiveTableInputFormat(
 			JobConf jobConf,
 			CatalogTable catalogTable,
-			List<HiveTablePartition> partitions) {
+			List<HiveTablePartition> partitions,
+			int[] projectedFields) {
 		super(jobConf.getCredentials());
 		checkNotNull(catalogTable, "catalogTable can not be null.");
 		this.partitions = checkNotNull(partitions, "partitions can not be null.");
 
 		this.jobConf = new JobConf(jobConf);
 		this.partitionColNames = catalogTable.getPartitionKeys();
-		rowArity = catalogTable.getSchema().getFieldCount();
+		int rowArity = catalogTable.getSchema().getFieldCount();
+		fields = projectedFields != null ? projectedFields : IntStream.range(0, rowArity).toArray();
```

Comment:
Let `projectedFields` never null?

Reply:
Since we've agreed to let projectedFields be null in HiveTableSource, I'll leave this unchanged.